snippets/varia/heatmaply/heatmaply.Rmd

```diff
@@ -0,0 +1,214 @@
+  dat.summary <- dat[, list(
+    median.val = median(get(expr_values)),
+    mean.val = mean(get(expr_values)),
+    cellspercluster = .N),
+    by = c('channel', 'cluster')]
+
+  # Decast the summarized data and convert to a matrix. Currently only the median values are returned
+  hm.cell <- dcast.data.table(dat.summary, formula='cluster ~ channel', value.var='median.val')
+  hm.clusters <- hm.cell$cluster
+  hm.cell <- as.matrix(hm.cell[, -1, with=FALSE])
+  
+  # Add rownames
+  rownames(hm.cell) <- hm.clusters
+
+  # Return the median values
+  return(as.matrix(hm.cell))
+}
+```
+
+## Run the `summarizeHeatmap` function
+```{r run-summarizeHeatmap, message=FALSE}
+# Select the relevant channels
+good.channels <- rownames(sce)[! rownames(sce) %in% c('H3', 'PD-1', 'cPARP', 'Ir191', 'Ir193')]
+
+hm <- summarizeHeatmap(sce,
+                       expr_values = 'counts',
+                       cluster = 'CellType',
+                       channels = good.channels)
+```
+
+## Expression heatmap
+Displays normalized, scaled or percentized counts per cluster and channel  
+Normalize: https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html#normalize  
+Scale: https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html#scale  
+Percentize: https://cran.r-project.org/web/packages/heatmaply/vignettes/heatmaply.html#percentize  
+```{r expression-heatmap-normalize}
+# Normalize
+heatmaply(heatmaply::normalize(hm))
+
+# Scale
+heatmaply(hm, scale="column")
+
+# Percentize
+heatmaply(heatmaply::percentize(hm))
+```
+
+# Using asinh-transformed counts
+```{r}
+# Caculate transformed counts
+assay(sce, "exprs") <- asinh(counts(sce) / 1)
+
+# Summarize the data
+hm.exprs <- summarizeHeatmap(sce,
+                             expr_values = 'exprs',
+                             cluster = 'CellType',
+                             channels = good.channels)
+
+# Plot the heatmap
+heatmaply(heatmaply::normalize(hm.exprs))
+```
+
+# Heatmap with other variables
+Any other variable in `colData(sce)` can be passed to `summarizeHeatmap`.  
+Example with donors (`case`).
+```{r}
+# Summarize the data
+hm.cases <- summarizeHeatmap(sce,
+                             expr_values = 'counts',
+                             cluster = 'case',
+                             channels = good.channels)
+
+# Plot the heatmap
+heatmaply(heatmaply::normalize(hm.cases))
+```
+
+
+
+# Add row-side colors
+Here, row side colors are defined in function of the cell types and cell categories (defined in `colData(sce)$CellType` and `colData(sce)$CellCat`).
+```{r row-side-colors}
+# Get cell types and cell categories
+rsc <- unique(data.frame(`Cell type`= colData(sce)$CellType,
+                         `Cell category` = colData(sce)$CellCat))
+
+# Plot the heatmap
+heatmaply(heatmaply::normalize(hm), RowSideColors = rsc)
+```
+
+
+## Correlation heatmap
+Displays correlation between channels
+```{r correlation-heatmap}
+heatmaply_cor(cor(hm))
```

Comment:
I would always specify the `method` argument when calculating the correlation - sometimes spearman correlation is preferred over pearson

Reply:
Ah, I meant specifying the method within the `cor(...)` function: `cor(..., method = "pearson")`. I'm not sure what exactly the `distfun` argument is doing.

Reply:
And also making sure that the same method is set in the `cor.test` call.

Reply:
Oh I see. Fixed it now.